rackcity/models/asset.py

```diff
@@ -161,7 +182,18 @@ class Asset(AbstractAsset):
         ITModel, on_delete=models.CASCADE, verbose_name="related model",
     )
     rack = models.ForeignKey(
-        Rack, on_delete=models.CASCADE, verbose_name="related rack",
+        Rack,
+        on_delete=models.CASCADE,
+        verbose_name="related rack",
+        null=True,
+        blank=True,
+    )
+    chassis = models.ForeignKey(
+        "self",
+        on_delete=models.PROTECT,
```

Comment:
shouldn't the on_delete behavior for chassis and rack be consistent?

Reply:
You shouldn't be able to delete either a rack or a chassis that has assets on it.  We check this manually on rack (and might end up checking it manually on chassis, likely for a better error message - I have a task to track this) but to be safe since I haven't implemented that check yet, I chose `PROTECT` so that the database won't allow a chassis with blades to be deleted.  We didn't know what `on_delete` did when we wrote it for Racks